Scripts/TileMap.gd

```diff
@@ -1,43 +1,84 @@
 extends TileMap
 
-signal proportions(tile_scale, grid_size,tile_size)
+signal proportions(grid_data)
+
+
 #this is the size of each tile
-@export var our_tile_size = Vector2i(16,16)
+@export var our_tile_size = Vector2(16,16)
 
 #how many tiles the grid should be
-@export var our_grid_size = Vector2i(12,12)
+@export var our_grid_size = Vector2(12,12)
+
+#the shape of the grid
+#grid_shape will return selected index
+@export_enum("square","cricle") var grid_shape : int
 
 # Called when the node enters the scene tree for the first time.
 func _ready():
 	create_grid()
 	
-
-
-#creates grid that that scalse with tile size and grid size
+	
+func _process(delta):
+	pass
+	
+	
+#creates grid that that scales with tile size and grid size
 func create_grid():
-	var grid = self
-	var base_tile_size = Vector2i(16,16)
-	var base_grid_size = Vector2i(12,12)
+	var grid_data : Dictionary
+	var selected_grid_shape : String
+	var base_tile_size : Vector2
+	var base_grid_size : Vector2
+	var grid_size_in_units : Vector2
+	var tile_center_offset : Vector2
+	var center_of_grid : Vector2
```

Comment:
Colons should be next to the variable name; see style guide (also everyone ever)

Reply:
no